### PR TITLE
Introduce string_utils for string functions

### PR DESCRIPTION
**Pull request by mtuncer**
Adds stringToInt and stringToUint functions, and replaces existing sscanf/strtol usages with this functions

### CODE REVIEW

File: src/bin/pg_autoctl/string_utils.c [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+ * representation of the integer.
+ */
+IntString
+intToString(int64_t number)
+{
+	IntString intString;
+
+	intString.intValue = number;
+
+	pg_snprintf(intString.strValue, INTSTRING_MAX_DIGITS, "%" PRId64, number);
```
Comment (JelteF) [inline src/bin/pg_autoctl/string_utils.c]: Let's keep this sprintf, beacuse it should be changed to `sformat` later.

```suggestion
	sprintf(intString.strValue, INTSTRING_MAX_DIGITS, "%" PRId64, number);
```

File: src/bin/pg_autoctl/string_utils.c [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+/*
+ * converts given string to 64 bit integer value.
+ * returns 0 upon failure and sets error flag
+ */
+int64_t
+stringToInt(const char *str, bool *error)
+{
+	char *endptr;
+	bool foundError = false;
+	int64_t number;
```
Comment (JelteF) [inline src/bin/pg_autoctl/string_utils.c]: This should be `long long int` and later it should be checked that number doesn't overflow/underflow int64 (same for function below). See this PR for my fix on the Citus versions:
https://github.com/citusdata/citus/pull/3602/files

File: src/bin/pg_autoctl/string_utils.c [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+ * returns 0 upon failure and sets error flag
+ */
+bool
+stringToInt(const char *str, int *number)
+{
+	char *endptr;
+	bool success = true;
+	long long int n;
+
+	if (number == NULL)
```
Comment (JelteF) [inline src/bin/pg_autoctl/string_utils.c]: should check str for null too

File: src/bin/pg_autoctl/string_utils.c [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+		return false;
+	}
+
+	errno = 0;
+
+	n = strtoll(str, &endptr, 10);
+
+	if (str == endptr)
+	{
+		success = false;
```
Comment (JelteF) [inline src/bin/pg_autoctl/string_utils.c]: This success variable is not really needed. It makes the code easier if we simply return early.
```suggestion
		return false;
```

File: src/bin/pg_autoctl/string_utils.c [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	return number;
+}
+
+
+/*
+ * converts given string to 64 bit unsigned integer value.
+ * returns 0 upon failure and sets error flag
+ */
+uint64_t
+stringToUInt(const char *str, bool *error)
```
[the remark this answers is not in the document]
Comment (mtuncer) [reply src/bin/pg_autoctl/string_utils.c]: did those with other stringToXXX friends

File: src/bin/pg_autoctl/string_utils.c (hunk 2 above)
Comment (mtuncer) [reply src/bin/pg_autoctl/string_utils.c]: ok

File: src/bin/pg_autoctl/string_utils.c (hunk 1 above)
Comment (mtuncer) [reply src/bin/pg_autoctl/string_utils.c]: done

File: src/bin/pg_autoctl/pgsql.c [hunk 6]
```diff
@@ -1247,7 +1245,7 @@ hostname_from_uri(const char *pguri,
 			if (option->val)
 			{
 				/* we expect a single port number in a monitor's URI */
-				*port = atoi(option->val);
+				stringToInt(option->val, port);
```
Comment (JelteF) [inline src/bin/pg_autoctl/pgsql.c]: Maybe check for error here

File: src/bin/pg_autoctl/pgsetup.c [hunk 7]
```diff
@@ -524,19 +544,14 @@ read_pg_pidfile(PostgresSetup *pgSetup, bool pg_is_not_running_is_ok)
 
 		if (lineno == LOCK_FILE_LINE_PORT)
 		{
-			sscanf(line, "%hu", &pgSetup->pidFile.port);
+			stringToUShort(line, &pgSetup->pidFile.port);
```
Comment (JelteF) [inline src/bin/pg_autoctl/pgsetup.c]: Maybe check for error here

File: src/bin/pg_autoctl/string_utils.c (hunk 3 above)
Comment (mtuncer) [reply src/bin/pg_autoctl/string_utils.c]: done

File: src/bin/pg_autoctl/string_utils.c (hunk 4 above)
Comment (mtuncer) [reply src/bin/pg_autoctl/string_utils.c]: changed

File: src/bin/pg_autoctl/pgsetup.c (hunk 7 above)
Comment (mtuncer) [reply src/bin/pg_autoctl/pgsetup.c]: added

File: src/bin/pg_autoctl/pgsql.c (hunk 6 above)
Comment (mtuncer) [reply src/bin/pg_autoctl/pgsql.c]: checked

File: src/bin/pg_autoctl/string_utils.c [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+	if (str == endptr)
+	{
+		foundError = true;
+	}
+	else if (errno != 0)
+	{
+		foundError = true;
+	}
+	else if (errno == 0 && str && *endptr != '\0')
```
[the remark this answers is not in the document]
Comment (mtuncer) [reply src/bin/pg_autoctl/string_utils.c]: ok

**Pull request merged**